collections/README.md

```diff
@@ -0,0 +1,1052 @@
+- module params will become a `collections.Item`
+- the `AllowList` is a `collections.KeySet`
+
+We don't want a collection to write over the state of the other collection so we pass it a prefix, which defines a storage
+partition owned by the collection.
+
+If you already built modules, the prefix translates to the items you were creating in your ``types/keys.go`` file, example: https://github.com/cosmos/cosmos-sdk/blob/main/x/feegrant/key.go#L27
+
+your old:
+```go
+var (
+	// FeeAllowanceKeyPrefix is the set of the kvstore for fee allowance data
+	// - 0x00<allowance_key_bytes>: allowance
+	FeeAllowanceKeyPrefix = []byte{0x00}
+
+	// FeeAllowanceQueueKeyPrefix is the set of the kvstore for fee allowance keys data
+	// - 0x01<allowance_prefix_queue_key_bytes>: <empty value>
+	FeeAllowanceQueueKeyPrefix = []byte{0x01}
+)
+```
+becomes:
+```go
+var (
+	// FeeAllowanceKeyPrefix is the set of the kvstore for fee allowance data
+	// - 0x00<allowance_key_bytes>: allowance
+	FeeAllowanceKeyPrefix = collections.NewPrefix(0)
+
+	// FeeAllowanceQueueKeyPrefix is the set of the kvstore for fee allowance keys data
+	// - 0x01<allowance_prefix_queue_key_bytes>: <empty value>
+	FeeAllowanceQueueKeyPrefix = collections.NewPrefix(1)
+)
+```
+
+#### Rules
+
+``collections.NewPrefix`` accepts either `uint8`, `string` or `[]bytes` it's good practice to use an always increasing `uint8`for disk space efficiency.
+
+A collection **MUST NOT** share the same prefix as another collection in the same module, and a collection prefix **MUST NEVER** start with the same prefix as another, examples:
+
+```go
+prefix1 := collections.NewPrefix("prefix")
+prefix2 := collections.NewPrefix("prefix") // THIS IS BAD!
+```
+
+```go
+prefix1 := collections.NewPrefix("a")
+prefix2 := collections.NewPrefix("aa") // prefix2 starts with the same as prefix1: BAD!!!
+```
+### Human-Readable Name
+
+The third parameter we pass to a collection is a string, which is a human-readable name.
+It is needed to make the role of a collection understandable by clients who have no clue about
+what a module is storing in state.
+
+#### Rules
+
+Each collection in a module **MUST** have a unique humanised name.
+
+## Key and Value Codecs
+
+A collection is generic over the type you can use as keys or values.
+This makes collections dumb, but also means that hypothetically we can store everything
+that can be a go type into a collection. We are not bounded to any type of encoding (be it proto, json or whatever)
+
+So a collection needs to be given a way to understand how to convert your keys and values to bytes.
+This is achieved through ``KeyCodec`` and `ValueCodec`, which are arguments that you pass to your
+collections when you're instantiating them using the ```collections.NewMap/collections.NewItem/...```
+instantiation functions.
+
+NOTE: Generally speaking you will never be required to implement your own ``Key/ValueCodec`` as
```

Comment:
Given it will be eventually displayed on docs.cosmos.network, we should adopt the syntax to make it more readable.

Check-out: https://docusaurus.io/docs/next/markdown-features/admonitions